### PR TITLE
Rename UnwrapJson to Wrapped

### PR DESCRIPTION
**Pull request by serj-lotutovici**
* Rename annotation and respective adapter.
* Introduce Wrapped.Factory for simple creation (for non-Retrofit use).

Inspired by #13 

/cc @vanniktech

### CODE REVIEW

**Comment by serj-lotutovici** [comment]
Codecov is complaining because the Factory constructor is not tested :) 

```
private Factory() {
      throw new AssertionError("No instances.");
}
```

**Comment by vanniktech** [comment]
I always test private constructors using [this](https://github.com/pushtorefresh/java-private-constructor-checker)

File: README.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ Full List of Adapters
 Moshi-Lazy-Adapters contains the following json adapters:
 * **SerializeNullsJsonAdapter** - Instructs moshi to serialize a value even if it's `null`;
 * **FirstElementJsonAdapter** - Instructs moshi to retrieve only the first element of a list;
-* **UnwrapJsonAdapter** - Unwraps a json object under the specified path;
+* **WrappedJsonAdapter** - Unwraps a json object under the specified path;
```
Comment (vanniktech) [inline README.md]: It not only `unwraps` it also `wraps`.

**Comment by serj-lotutovici** [comment]
Thx @vanniktech for the hint.

**Pull request merged**